### DISCUSSION TITLE
make driver help st7789

### DISCUSSION THREAD

**Discussion by hansendw** [micropython/micropython#16492] [Libraries & Drivers]
i want to use the driver st7789. I went to the github and got the folder https://github.com/russhughes/st7789_mpy. I have this folder in my /Micropython folder. It's a Linux machine running Mint. I am uncertain what to do next. No doubt this is a simple procedure and I apologize for not knowing what to do next. I will use this driver for a micropython esp-32 board.

**Comment by peterhinch** [comment]
As an alternative you might like to look at [nano-gui](https://github.com/peterhinch/micropython-nano-gui) whcih has its own ST7789 driver. This will run on an ESP32.

**Reply by hansendw** [reply]
Thanks, but that doesn't help me. Im running Micropython on an esp32. I just need to have the display driver ST7789

**Reply by peterhinch** [reply]
The link you posted provides detailed installation instructions. Russ Hughes' driver is written in C so requires compiling as part of the firmware. This enables high speed operation. I would suggest learning to create and install a standard firmware build before following his excellent instructions.

The driver I suggested is written in Python and can be installed easily with mip. The trade-off is performance.

**Reply by hansendw** [reply]
The problem is that following the directions enclosed in his readme file, doesn't work. I am looking for the mechanics of installing a display driver to micropython, not just a "copy and paste" code.

**Reply by hansendw** [reply]
I found the answer
https://www.youtube.com/watch?v=hdWts-q1tMo&t=22s

**Reply by peterhinch** [reply]
Have you had success with the instructions in that video? I took a quick look. It calls for a Python driver which I can't find i the repo.
